packages/base/src/3dview/mainview.tsx

```diff
@@ -617,7 +619,7 @@ export class MainView extends React.Component<IProps, IStates> {
           this._edgeMaterials.push(el.material);
           if (selectedNames.includes(el.name)) {
             this._selectedMeshes.push(el as any as BasicMesh);
-            el.material.color = SELECTED_MESH_COLOR;
+            el.material.color = objColor;
```

Comment:
I would keep `SELECTED_MESH_COLOR` here, as it's the color of selected meshes (for highlighting).

Though I believe you may be missing another spot where to set the color. When we deselect objects we bring back the `DEFAULT_MESH_COLOR`. I would search through the codebase everywhere `DEFAULT_MESH_COLOR` is used and replace it with `objColor || DEFAULT_MESH_COLOR`

Reply:
Thanks a lot. I'm now able to give custom color to the box model. woohoo 

https://github.com/user-attachments/assets/710d8a37-85b6-490c-b7ff-1e53192f2250

However it is still returning to `DEFAULT_MESH_COLOR` when deselected at the moment, i'm looking into this assuming it is in [`_updateSelected`](https://github.com/jupytercad/JupyterCAD/blob/main/packages/base/src/3dview/mainview.tsx#L842) method, I'll make sure to ask again if I feel stuck. Thanks again :)